src/java/org/apache/cassandra/db/partitions/UnfilteredPartitionIterators.java

```diff
@@ -45,6 +45,7 @@ private UnfilteredPartitionIterators() {}
 
     public interface MergeListener
     {
+        public default boolean preserveOrder() { return true; }
```

Comment:
This needs a short javadoc/explanation.

`NOOP` below should be returning `false` for this.

Reply:
Updated NOOP and added javadoc comment